graphql_api/types/comparison/comparison.py

```diff
@@ -184,18 +184,30 @@ def resolve_flag_comparisons(
 @comparison_bindable.field("componentComparisons")
 @sync_to_async
 def resolve_component_comparisons(
-    comparison_report: ComparisonReport, info
+    comparison_report: ComparisonReport, info, filters=None
 ) -> List[ComponentComparison]:
     current_owner = info.context["request"].current_owner
     head_commit = comparison_report.commit_comparison.compare_commit
     components = components_service.commit_components(head_commit, current_owner)
+    list_components = comparison_report.commit_comparison.component_comparisons.all()
+
+    if filters and filters.get("components"):
+        components = [
+            component
+            for component in components
+            if component.name in filters["components"]
```

Comment:
Are component names guaranteed to be unique?

Reply:
good question i'll follow up on that

Reply:
Looking at https://docs.codecov.com/docs/components a component is unique by id, nevertheless i still think we want to filter by name, as the user is expecting name of components in the dashboard and they would expect the filter to match display names

that being said, i'll raise this with product/design to see if displaying the id in upcoming iteration would be helpful to have a more accurate components filter, thanks Rohit!